app/controllers/pets_controller.rb

```diff
@@ -5,6 +5,10 @@ class PetsController < ApplicationController
   # GET /pets.json
   def index
     @pets = Pet.all
+
+    if not @pets.nil?
```

Comment:
Did you ever get a situation when @pets were nil? If not, this check is redundant.